api/src/opentrons/protocol_engine/state/pipettes.py

```diff
@@ -376,15 +369,20 @@ def get_deck_point(self, pipette_id: str) -> Optional[DeckPoint]:
             return current_deck_point.deck_point
         return None
 
-    def get_aspirated_volume(self, pipette_id: str) -> float:
+    def get_aspirated_volume(self, pipette_id: str) -> Optional[float]:
         """Get the currently aspirated volume of a pipette by ID."""
         try:
-            return self._state.aspirated_volume_by_id[pipette_id]
+            aspirate_volume = self._state.aspirated_volume_by_id[pipette_id]
         except KeyError as e:
             raise errors.PipetteNotLoadedError(
                 f"Pipette {pipette_id} not found; unable to get current volume."
             ) from e
 
+        if pipette_id not in self._state.tip_volume_by_id.keys():
```

Comment:
Per comments above, I don't think checking for key presence is sufficient.

I think we should switch this to:

```suggestion
        if self._state.tip_volume_by_id.get(pipette_id) is None:
```